src/ILCompiler.Compiler/src/Compiler/LibraryRootProvider.cs

```diff
@@ -45,6 +45,8 @@ public void AddCompilationRoots(IRootingServiceProvider rootProvider)
                 // If this is not a generic definition, root all methods
                 if (!type.HasInstantiation)
                     RootMethods(type, "Library module method", rootProvider);
+
+                rootProvider.RootStaticBasesForType(type, "Library module type statics");
```

Comment:
Nit: I would fold this under the `!HasInstantiation` check instead of `RootStaticBases` silently doing nothing if you pass it a nonsensical input (a generic definition).